addons/stock/models/product.py

```diff
@@ -315,7 +315,13 @@ def _get_domain_locations_new(self, location_ids, company_id=False, compute_chil
             dest_loc_domain = dest_loc_domain + [('location_dest_id', operator, other_locations.ids)]
         usage = self._context.get('quantity_available_locations_domain')
         if usage:
-            stock_loc_domain = expression.AND([domain + loc_domain, [('location_id.usage', 'in', usage)]])
+            if 'transit' in usage:
+                usage.remove('transit')
+                default_loc_domain = expression.AND([loc_domain, [('location_id.usage', 'in', usage)]])
+                transit_loc_domain = expression.OR([default_loc_domain, [('location_id.usage', '=', 'transit')]])
+                stock_loc_domain = expression.AND([domain, transit_loc_domain])
```

Comment:
i don't get the whole if block :/

Reply:
Originally, stock_loc_model resulted in something like this:
```
['&', ('company_id', '=', company_id), '&', ('location_id.parent_path', '=like', '1/7/%'), ('location_id.usage', 'in', ['internal'])]`
```

With the part about the `parent_path`, I couldn't simply add `transit` is the usage part of the domain.

This block allows to reconstruct the domain like this:
```
['&', ('company_id', '=', company_id), '|', '&', ('location_id.parent_path', '=like', '1/7/%'), ('location_id.usage', 'in', ['internal']), ('location_id.usage', '=', 'transit')]
```